extras/check-spirv-generated.sh

```
@@ -0,0 +1,262 @@
+else
+  SPIRV_CHANGED=false
+  if echo "$CHANGED_FILES" | grep -q "^external/spirv-tools"; then
+    echo_info "Detected changes in external/spirv-tools"
+    SPIRV_CHANGED=true
+  fi
+  if echo "$CHANGED_FILES" | grep -q "^external/spirv-headers"; then
+    echo_info "Detected changes in external/spirv-headers"
+    SPIRV_CHANGED=true
+  fi
+fi
+
+if [ "$SPIRV_CHANGED" = false ]; then
+  echo_success "No changes to external/spirv-tools or external/spirv-headers - skipping check"
+  exit 0
+fi
+
+echo ""
+echo_info "Changes detected in SPIRV directories - verifying generated files are up-to-date..."
+echo ""
+
+# 2. Initialize submodules
+echo_info "Initializing SPIRV submodules..."
+git submodule update --init --recursive external/spirv-tools
+git submodule update --init --recursive external/spirv-headers
+
+# 3. Sync spirv-tools dependencies
+echo_info "Syncing spirv-tools dependencies..."
+cd external/spirv-tools
+
+if [ ! -f "utils/git-sync-deps" ]; then
+  echo_error "utils/git-sync-deps not found in external/spirv-tools"
+  exit 1
+fi
+
+# git-sync-deps needs to be run to get additional dependencies
+python3 utils/git-sync-deps
+
+cd ../..
+
+# 4. Build spirv-tools to generate files
+BUILD_DIR="external/spirv-tools/build-ci-check"
+echo_info "Building spirv-tools generation targets in $BUILD_DIR..."
+
+mkdir -p "$BUILD_DIR"
+
+# Configure CMake (suppress warnings from external project)
+cmake -Wno-dev -B "$BUILD_DIR" external/spirv-tools
+
+# Build only the generation targets (faster than full build)
+echo_info "Building generation targets (this may take a few minutes)..."
+cmake --build "$BUILD_DIR" \
+  --target spirv-tools-build-version \
+  --target core_tables \
+  --target extinst_tables
+
+# 5. Compare generated files
+echo_info "Comparing generated files..."
+TEMP_DIR=$(mktemp -d)
+
+# Copy generated files to temp directory
+cp "$BUILD_DIR"/*.inc "$TEMP_DIR/" 2>/dev/null || true
+cp "$BUILD_DIR"/*.h "$TEMP_DIR/" 2>/dev/null || true
```

why `|| true` needed?

---

Not needed, it unnecessarily masks errors. Fixed.